src/govuk/components/modal-dialogue/modal-dialogue.js

```diff
@@ -0,0 +1,186 @@
+import { nodeListForEach } from '../../common'
+import '../../vendor/polyfills/Element/prototype/classList'
+import '../../vendor/polyfills/Function/prototype/bind'
+import '../../vendor/polyfills/Event' // addEventListener and event.target normaliziation
+
+function ModalDialogue ($module) {
+  this.$module = $module
+  this.$dialogBox = $module.querySelector('dialog')
+  this.$container = document.documentElement
+
+  // Check for browser support
+  this.hasNativeDialog = 'showModal' in this.$dialogBox
+
+  // Allowed focussable elements
+  this.focussable = [
```

Comment:
There is an 'inert' feature we could consider referring to when reviewing the focus trap functionality in this proposal:

https://github.com/WICG/inert

It ships in chromium, I'm not sure if it's a proper standard yet but worth thinking about.

Reply:
Yeah sounds like a good enhancement.

You'd have to be _very_ careful and keep the modal outside `govuk-width-container` or whatever the page wrapper element was.

```html
<body>
    <div class="govuk-width-container" inert>…</div>
    <div data-module="govuk-modal-dialogue">…</div>
</body>
```

It could go horribly wrong if the modal was inside the `[inert]` wrapper.

Reply:
The early prototype version of the modal that GOV.UK based theirs on [used both the `inert` attribute and `aria-hidden`](https://github.com/hannalaakso/accessible-timeout-warning/blob/0991356bd5ceba342e810e494e567fa61c1fe9b9/src/govuk/components/timeout-warning/timeout-warning.js#L228) to make the `<main>` wrapper non-actionable and this tested well with screen reader users. The modal markup was outside `<main>` and below `<body>`. 

Colin does have a valid point: we need to be clear in the guidance about where the modal markup is placed in the page content.